### PR TITLE
feat: chart x-axis label rotation

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6327 

### Description:

Enables label rotation for the x-axis (non-flipped). We can add this for the other axes if necessary. I did some of the setup, but wanted to see if we found this the right approach. And if we add it fro the other labels we probably should think about the UI differently; here it is just a single number input. 

A couple notes:
- This moves the axis name in a way that *usually* looks good, but can end up pushing the name offscreen if the fields are long. We could either add field truncation (I played with this and have some ideas) or allow them to place the name. 
- When there are *a lot* of labels, some still get hidden. Theres an echarts bug about this. 

The code in the config file is a little tangled -- I think I made reasonable tradeoffs 😅, but I'm happy to make changes. I added a couple comments in this review to try to clarify.  

Example:
Screenshot 2024-01-18 at 19 24 38

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/AxesOptions.tsx [hunk 1]
```diff
@@ -165,6 +167,26 @@ const AxesOptions: FC<Props> = ({ itemsMap }) => {
                     }}
                 />
             </Group>
+            {!dirtyLayout?.flipAxes && (
```
Comment (magnew) [inline packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/AxesOptions.tsx:170]: Doesn't show when axes are flipped

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 2]
```diff
@@ -749,7 +749,25 @@ const calculateWidthText = (text: string | undefined): number => {
     return width;
 };
 
-const getEchartAxis = ({
+const getLongestLabel = ({
```
Comment (magnew) [inline packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:752]: Moved here from inlines below

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 3]
```diff
@@ -1276,12 +1331,12 @@ const useEchartsCartesianConfig = (
         );
     }, [validCartesianConfig, resultsData, itemsMap, pivotDimensions]);
 
-    const axis = useMemo(() => {
+    const axes = useMemo(() => {
```
Comment (magnew) [inline packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:1334]: Just a rename for clarity

**Review by rephus — APPROVED** [review]
Tested and reviewed, it works great :+1:

**Pull request merged**